cmd/libsnap-confine-private/mountinfo-test.c

```diff
@@ -156,6 +156,28 @@ static void test_parse_mountinfo_entry__many_tags()
 	g_assert_null(entry->next);
 }
 
+static void test_parse_mountinfo_entry__empty_source()
+{
+	const char *line =
+	    "304 301 0:45 / /snap/test-snapd-content-advanced-plug/x1 rw,relatime - tmpfs  rw";
```

Comment:
The two spaces between 'tmpfs' and 'rw': is this _guaranteed_ to denote an empty source? Ie, is it an implementation artifact or is it part of the specification for mountinfo? This seems scarily brittle at first glance.

Reply:
It is just how the kerne is impemented. Reading the specification it is neither violating it nor explicitly explained that it behaves this way. IMO this is a kernel bug but the way countess kernel interfaces are defined it doesn’t stand out. I for one would love to see standardized serializer but this is unlikely to happen anytime soon.

I looked at how this is implemented and it is a %s with some escape logic for spaces and other characters.